spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -132,6 +132,9 @@ protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap opti
     optionsMap.putAll(options.asCaseSensitiveMap());
     optionsMap.put(CatalogProperties.APP_ID, SparkSession.active().sparkContext().applicationId());
     optionsMap.put(CatalogProperties.USER, SparkSession.active().sparkContext().sparkUser());
+    optionsMap.putIfAbsent(
+        CatalogProperties.WAREHOUSE_LOCATION,
+        SparkSession.active().sqlContext().conf().warehousePath());
```

Comment:
of course, I agree that for HadoopCatalog accessed separately, if no warehouse is set, an error should be thrown.
but i think `spark.sql.warehouse.dir` should have same behavior for all catalogs in spark,  so this behavior i think is ok.

if you are still concerned about the possibility of breaking the previous behavior, I can change this logic to only take effect for HiveCatalog. I would like to hear your opinion.

Reply:
I think it's fine to use that behavior everywhere, I don't really think we would break anyone's old flows since they are already are specifying the location at the catalog property level. Allowing this as a fall back seems pretty reasonable to me.

Reply:
HiveCatalog will use CatalogProperties.WAREHOUSE_LOCATION to set to HiveConf.ConfVars.METASTOREWAREHOUSE.varname during initialization, so I think it is better to use CatalogProperties.WAREHOUSE_LOCATION here.